tools/cli/workflow.go

```diff
@@ -82,50 +82,24 @@ func newWorkflowCommands() []cli.Command {
 			Name:    "signal",
 			Aliases: []string{"s"},
 			Usage:   "signal a workflow execution",
-			Flags: []cli.Flag{
-				cli.StringFlag{
-					Name:  FlagWorkflowIDWithAlias,
-					Usage: "WorkflowID",
-				},
-				cli.StringFlag{
-					Name:  FlagRunIDWithAlias,
-					Usage: "RunID",
-				},
-				cli.StringFlag{
-					Name:  FlagNameWithAlias,
-					Usage: "SignalName",
-				},
-				cli.StringFlag{
-					Name:  FlagInputWithAlias,
-					Usage: "Input for the signal, in JSON format.",
-				},
-				cli.StringFlag{
-					Name:  FlagInputFileWithAlias,
-					Usage: "Input for the signal from JSON file.",
-				},
-			},
+			Flags:   getFlagsForSignal(),
 			Action: func(c *cli.Context) {
 				SignalWorkflow(c)
 			},
 		},
+		{
+			Name:  "signalwithstart",
+			Usage: "signal with start a workflow execution",
```

Comment:
how about : Signal the workflow if an existing run is running, or start a workflow if a new run can be started based on IDResuePolicy, or ,otherwise return an error when attempting to start a new run.